### ISSUE TITLE
Nested unordered lists throw an accessibility warning

### ISSUE THREAD

**Issue by anthony-liddle** [microsoft/roosterjs#2999]
**Describe the bug**
Nested unordered lists trigger an accessibility error when using the Accessibility Insights extension and running a FastPass. 

This issue exists in both the latest version of RoosterJS, and in the version we are using for the text editor in our component library (8.61.0).

When adding a nested unordered list, the HTML renders as:
```
<ul>
    <li>first</li>
    <li>second</li>
    <ul>
        <li>indent one</li>
        <li>indent two</li>
    </ul>
    <li>third</li>
</ul>
```

[According to W3](https://www.w3.org/WAI/tutorials/page-structure/content/#nested-lists), nested lists should be formatted as:
```
<ul>
    <li>first</li>
    <li>second
        <ul>
            <li>indent one</li>
            <li>indent two</li>
        </ul>
    </li>
    <li>third</li>
</ul>
```
The above will not trigger a warning using the FastPass in the Accessibility Insights extension.

Here is the error from FastPass for the latest version of RoosterJS:
```
Title: WCAG 1.3.1: Ensure that lists are structured correctly (ul)
Tags: Accessibility, WCAG 1.3.1, list

Issue: Ensure that lists are structured correctly (list - https://accessibilityinsights.io/info-examples/web/list)

Target application: RoosterJs Demo Site - https://microsoft.github.io/roosterjs/index.html#

Element path: #RoosterJsContentDiv > ul

Snippet: <ul data-editing-info="{&quot;applyListStyleFromLevel&quot;:true}" style="list-style-type: disc;">

Related paths: 
  ul > ul

How to fix: 
Fix all of the following:
  List element has direct children that are not allowed: ul

Environment: Microsoft Edge version 135.0.0.0

====

This accessibility issue was found using Accessibility Insights for Web 2.46.0 (axe-core 4.10.2), a tool that helps find and fix accessibility issues. Get more information & download this tool at http://aka.ms/AccessibilityInsights.
```

**To Reproduce**
1. Go to the [Rooster JS Demo page](https://microsoft.github.io/roosterjs/index.html#)
2. Click on the Unordered List menu button
3. Type a couple bullet points
4. Hit the `tab` key to add a nested unordered list
5. Type a couple bullet points
6. See the error that the rendered HTML adds a `ul` as a sibling of the `li` elements instead of as a child of a `li` element.

<img width="529" alt="Image" src="https://github.com/user-attachments/assets/cc747b37-c957-459b-a14b-43718683a312" />

<img width="671" alt="Image" src="https://github.com/user-attachments/assets/cc440c10-e4f0-4959-b776-a501d42d7488" />

**Device Information**
 - OS: Mac
 - Browser: Edge
 - Version: 135
